auslib/admin/views/forms.py

```diff
@@ -78,6 +80,45 @@ def coercer(value):
     return coercer
 
 
+def operator_validator():
+    log.debug('starting in operator_validator for buildID')
+
+    def _validator(form, field):
+        log.debug('starting in operator_validator: field.data is %s' % field.data)
+        # empty input is fine
+        if field.data is None:
+            return
+        try:
+            op, operand = get_op(field.data)
+            log.debug('Got (%s, %s) from get_op', op, operand)
+        except TypeError:
+            # get_op field returns None if no operator or no match, can't be unpacked
+            raise ValidationError('Invalid input for %s' % field.label)
+    return _validator
+
+
+def version_validator():
+    log.debug('starting in version_validator for version')
+
+    def _validator(form, field):
+        log.debug('starting in version_validator: field.data is %s' % field.data)
+        # empty input
+        if field.data is None:
+            return
+        try:
+            op, operand = get_op(field.data)
+            version = MozillaVersion(operand)
+        except ValueError:
+            raise ValidationError("Couldn't parse version for %s" % field.label)
+        except:
+            raise ValidationError('Invalid input for %s' % field.label)
+        # MozillaVersion doesn't error on empty strings
+        if not hasattr(version, 'version'):
+            raise ValidationError("Couldn't parse the version for %s" % field.label)
```

Comment:
The error messages for both of these could be improved, this is what they look like now:
http://people.mozilla.org/~bhearsum/sattap/64b6797c.png

They should probably say something more precise than "invalid input" when the operator parsing fails, and it looks like field.label isn't the best thing to use here.

Once these are fixed, I'll get this merged.